chain/indexer/src/handlers.rs

```diff
@@ -433,10 +446,17 @@ where
                     );
 
                     self.db.upsert_channel(tx.into(), new_channel).await?;
-                    new_channel
+                    Some(new_channel)
                 };
 
-                Ok(Some(ChainEventType::ChannelOpened(channel)))
+                match channel {
+                    Some(c) => Ok(Some(ChainEventType::ChannelOpened(c))),
+                    None => {
+                        // Should not happen as such event should already have trigered a return Ok(None) before
+                        error!(%source, %destination, %channel_id, "observed channel open event for a channel that does not exist");
+                        Err(CoreEthereumIndexerError::ChannelDoesNotExist)
+                    }
+                }
```

Comment:
It seems to me, that if you apply `ok_or(...)?` on line 434, and change line 449 to just `new_channel`, you can drop this entire `match` statement and just return `Ok(Some(ChainEventType::ChannelOpened(channel)))` as it was before.